### PR TITLE
fix(packages/agent) Added lazy queue 

### LINKED ISSUE #423: Error: Unable to get property 'autoLink' of undefined or null reference
https://sentry.io/uncovertruth/agent/issues/497931811/

```
None: Unable to get property 'autoLink' of undefined or null reference
  at None (agent.d.js:1:1)
```

### PR DESCRIPTION
**Pull request by 9renpoto**
# TL;DR

Closes #423

### CODE REVIEW

File: packages/agent/test/global.test.ts [hunk 1]
```diff
@@ -14,7 +14,6 @@ describe('global async', () => {
     inject('', { [NAMESPACE]: GLOBAL_NAME })
   })
 
-  const w: any = window
```
Comment (9renpoto) [inline packages/agent/test/global.test.ts]: remove not replaced cache

File: packages/agent/src/agent.ts [hunk 2]
```diff
@@ -96,9 +96,12 @@ export default class Agent {
 
   provide (name: string, pluginConstructor: PluginConstructor) {
     this[PLUGINS][name] = pluginConstructor
+    console.warn(this[PLUGINS], 'provide')
```
Comment (9renpoto) [inline packages/agent/src/agent.ts]: oops debug print

File: packages/agent/src/global.ts [hunk 3]
```diff
@@ -47,24 +51,42 @@ const execute = (
...
 
-const isCreateCmd = (isEqual: boolean) => ({ 0: cmd }: Arguments): boolean =>
-  isEqual === (cmd === CREATE)
+const obj2array = (args: object) => [].map.call(args, (x: any) => x)
 
-export default function (Agent: any) {
+export default function (
+  Agent: any,
+  cnt: { [key: string]: number },
+  agents: { [key: string]: any }
```
Comment (9renpoto) [inline packages/agent/src/global.ts]: global cache scope

File: packages/agent/src/global.ts [hunk 4]
```diff
@@ -47,24 +51,42 @@ const execute = (
...
+      lazyStack[cmd] = 0
+    }
+
+    setTimeout(() => {
+      const res = execute(Agent, agents)(cmd, ...args)
+      if (!res) {
+        lazyStack[cmd]++
+        lazyStack[cmd] < 5
+          ? q.push(argsObject)
+          : warning(`exezcute timeout: ${cmd}`)
```
Comment (9renpoto) [inline packages/agent/src/global.ts]: 😰

File: packages/agent/src/agent.ts [hunk 5]
```diff
@@ -109,8 +110,13 @@ export default class Agent {
     return false
   }
 
-  run (pluginName: string, methodName: string, ...args: any[]) {
-    this.plugins[pluginName][methodName](...args)
+  run (pluginName: string, methodName: string, ...args: any[]): boolean {
+    const p = this[PLUGINS][pluginName]
+    if (p && p[methodName]) {
+      const res = p[methodName](...args)
+      return typeof res === 'undefined' ? true : !!res
```
Comment (shoota) [inline packages/agent/src/agent.ts]: need `typeof`? it can check `res===undefined`.

File: packages/agent/src/global.ts [hunk 6]
```diff
@@ -47,24 +50,41 @@ const execute = (
...
+      const res = execute(Agent, agents)(cmd, ...args)
+      if (!res) {
+        lazyStack[cmd]++
+        lazyStack[cmd] < 5
+          ? q.push(argsObject)
+          : warning(`execute timeout: ${cmd}`)
+      }
+      const [next, ...queue] = obj2array(q)
+      applyQueue(next, queue)
+    }, lazyStack[cmd] * lazyStack[cmd] * 100)
```
Comment (9renpoto) [inline packages/agent/src/global.ts]: wait 5 sec

**Pull request merged**